mango-cli/src/main/scala/org/bdgenomics/mango/cli/VizReads.scala

```diff
@@ -549,6 +555,106 @@ class VizServlet extends ScalatraServlet {
     }
   }
 
+  post("/ga4gh/variants/search") {
+    val jsonPostString = request.body
+    val searchVariantsRequest: SearchVariantsRequestGA4GH = net.liftweb.json.parse(jsonPostString)
+      .extract[SearchVariantsRequestGA4GH]
+
+    if (!VizReads.materializer.variantContextExist)
+      VizReads.errors.notFound
+    else {
+      val viewRegion = ReferenceRegion(searchVariantsRequest.referenceName,
+        searchVariantsRequest.start.toLong,
+        VizUtils.getEnd(searchVariantsRequest.end.toLong,
+          VizReads.globalDict(searchVariantsRequest.referenceName)))
+
+      //currently only supporting a single call set id in the query, while API supports a list of call set ids
+      val key = searchVariantsRequest.callSetIds.head
+      contentType = "json"
+
+      val dictOpt = VizReads.globalDict(viewRegion.referenceName)
+
+      if (dictOpt.isDefined) {
+        var results: Option[String] = None
+
+        val binning: Int = 1
+
+        VizReads.variantsWait.synchronized {
+          if (!VizReads.variantsIndicator.region.contains(viewRegion) || binning != VizReads.variantsIndicator.resolution) {
+            val expanded = VizReads.expand(viewRegion)
+            VizReads.variantsCache = VizReads.materializer.getVariantContext().get.getJson(expanded,
+              VizReads.showGenotypes,
+              binning)
+            VizReads.variantsIndicator = VizCacheIndicator(expanded, binning)
+          }
+        }
+
+        // filter data overlapping viewRegion and stringify
+        val data: Array[VariantContext] = VizReads.variantsCache.get(key).getOrElse(Array.empty)
+          .filter(z => { ReferenceRegion(z.variant.variant).overlaps(viewRegion) })
+
+        results = Some(VizReads.materializer.getVariantContext().get.stringifyGA4GH(data))
+
+        if (results.isDefined) {
+          // extract variants only and parse to stringified json
+          Ok(results.get)
+        } else VizReads.errors.noContent(viewRegion)
+      } else VizReads.errors.outOfBounds
+    }
+  }
+
+  post("/ga4gh/variants/search/bin") {
+
+    val jsonPostString = request.body
+
+    val searchVariantsRequest: SearchVariantsRequestGA4GHBinning = net.liftweb.json.parse(jsonPostString)
+      .extract[SearchVariantsRequestGA4GHBinning]
```

Comment:
The above is request objectr specifying the query, not the responses.
The response actually is generic being a GA4GH `Feature`
https://github.com/jpdna/mango/blob/ga4gh7/mango-cli/src/main/scala/org/bdgenomics/mango/cli/VizReads.scala#L646
and can be same for reads, vars, etc.

The request must be type specific because the case class used to parse JSON must have the field name like "var_set_id" or "read_set_id" as required by GA4GH API.   

Finally - I do think I may be able to get rid now of the "binned" version of the request, so just have one request object per type that works for both binned and not binned - I'm not sure why I made both earlier.